### PR TITLE
Update template to be using context

### PR DESCRIPTION
**Pull request by ulziibay**
# Jira

[SECNG-1826](https://clever.atlassian.net/browse/SECNG-1826)

# Overview
Use CircleCI contexts defined and materialized in https://clever.atlassian.net/browse/SECNG-1827.

# Testing
See https://github.com/Clever/init-service/pull/153
# Rollout

## Rollback (in the event of a problem)
- [ ] Rollback via dapple OR `ark rollback -e production {{.AppName}}`?
- [ ] Anything else?

# New repo setup

- [ ] Ensure you have `npm 6` installed and then run `npm install`. Commit the generated `package-lock.json`
- [ ] Populate the `HOST` and `PORT` env vars in [ark-config](https://github.com/Clever/ark-config)
- [ ] Set up Slack notifications for this app for your team if you'd like ([GitHub assignments and notifications](https://clever.atlassian.net/wiki/spaces/ENG/pages/888897571/GitHub+assignments))
- [ ] Tune your alarms if you'd like anything other than our default recommendations ([alarm best practices](https://clever.atlassian.net/wiki/spaces/~620990898/pages/904036784/Alarm+Best+Practices))
- [ ] Step through the repo's TODOs and prioritize them as you see fit. You can ignore TODOs suffixed with `(template-frontend)` as those are intended for the maintainers of the template, not you as a consumer of the template
- [ ] Adjust the cpu and max_mem values based on the needs of your application as the containers are killed if these values are exceeded
- [ ] If this app does not need to be canaried, you can disable that in the `deploy_config` section of the launch config
- [ ] If this app should be multi region in sso pods adjust the pod_config https://clever.atlassian.net/wiki/spaces/ENG/pages/370147335/launch.yml#launch.yml-PodConfig
- [ ] Delete this section from the PR template

[SECNG-1825]: https://clever.atlassian.net/browse/SECNG-1825?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[SECNG-1826]: https://clever.atlassian.net/browse/SECNG-1826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,7 +48,8 @@ jobs:
       - setup_remote_docker
       - run: ../ci-scripts/circleci/docker-publish $DOCKER_USER $DOCKER_PASS "$DOCKER_EMAIL" $DOCKER_ORG
       - run: ../ci-scripts/circleci/catapult-publish $CATAPULT_URL $CATAPULT_USER $CATAPULT_PASS {{.AppName}}
-      # TODO: Uncomment this if you'd like to serve assets via the CDN
+      # TODO: Uncomment this if you'd like to serve assets via the CDN.
+      # TODO: add `context: aws-s3-upload` to publish step so that aws role based authentication works out
```
Comment (mcab) [inline .circleci/config.yml:52]: Can we add this already to uncomment for S3?

**Comment by mcab** [comment]
Some additional things wrapped up here, too.

```
% npm i
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated @stylelint/postcss-markdown@0.36.2: Use the original unforked package instead: postcss-markdown
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm WARN deprecated redux-devtools-extension@2.13.9: Package moved to @redux-devtools/extension.
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated @stylelint/postcss-css-in-js@0.37.3: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

added 1504 packages, and audited 1505 packages in 26s

219 packages are looking for funding
  run `npm fund` for details

3 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

**Comment by mcab** [comment]
Oh, the context changes work, too:

```
      - publish:
          context:
            - [ecr private]
            # If you want to upload files to S3 while publishing the app, uncomment this line.
            # - [s3 upload]
          requires:
            - build
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -84,6 +84,10 @@ workflows:
           requires:
             - build
       - publish:
+          context:
+            - {{publishEcr .}}
+            # If you want to upload files to S3 while publishing the app, uncomment this line.
+            # - {{publishS3 .}}
```
Comment (ulziibay) [inline .circleci/config.yml:90]: 👍

Comment (mcab) [reply .circleci/config.yml:90]: This may get exposed in public creations of frontend apps, but I can't think of a use-case for such.

**Review by ulziibay — COMMENTED** [review]
Looks good! i cannot approve due to Github limitation

**Pull request merged**